tutor/src/components/scores/index.cjsx

```diff
@@ -25,209 +26,83 @@ Scores = React.createClass
     courseId: React.PropTypes.string.isRequired
     isConceptCoach: React.PropTypes.bool.isRequired
 
-  mixins: [ResizeListenerMixin]
-
   getInitialState: ->
     sortedPeriods = CourseStore.getPeriods(@props.courseId)
-
     period_id: _.first(sortedPeriods).id
     periodIndex: 1
     sortIndex: 0
-    tableWidth: 0
-    tableHeight: 0
-    colDefaultWidth: 160
-    colSetWidth: 160
-    colResizeWidth: 160
-    colResizeKey: 0
     sort: { key: 'name', asc: true, dataType: 'score' }
-    # index of first column that contains data
-    firstDataColumn: 2
     displayAs: 'percentage'
 
-  componentDidMount: ->
-    @sizeTable()
-
-  _resizeListener: ->
-    @sizeTable()
-
-  sizeTable: ->
-    @setState({tableWidth: @tableWidth(), tableHeight: @tableHeight()})
-
-  tableWidth: ->
-    windowEl = @_getWindowSize()
-    tableContainer = React.findDOMNode(@refs.tableContainer) #.course-scores-container
-    style = tableContainer.currentStyle or window.getComputedStyle(tableContainer)
-    padding = parseInt(style.paddingLeft) + parseInt(style.paddingRight)
-    tableContainerWidth = tableContainer.clientWidth - padding
-    tableHorzSpacing = document.body.clientWidth - tableContainerWidth
-    # since table.clientWidth returns 0 on initial load in IE, include windowEl as a fallback
-    Math.max(windowEl.width - tableHorzSpacing, tableContainerWidth)
-
-  tableHeight: ->
-    windowEl = @_getWindowSize()
-    table = React.findDOMNode(@refs.tableContainer)
-    bottomMargin = 140
-    windowEl.height - table.offsetTop - bottomMargin
-
-
+  componentWillMount:  -> @updateStudentData()
   changeSortingOrder: (key, dataType) ->
-    asc = if @state.sort.key is key then not @state.sort.asc else false
-    @setState(sort: {key, asc, dataType})
-
-  isSortingByData: ->
-    _.isNumber(@state.sort.key)
-
-  selectPeriod: (period) ->
-    newState = {period_id: period.id}
-    newState.sort = @state.sort if @isSortingByData()
-    @setState(newState)
-
-  setPeriodIndex: (key) ->
-    @setState({periodIndex: key + 1})
-
-
-  changeDisplayAs: (mode) ->
-    @setState(displayAs: mode)
-
-  percent: (num, total) ->
-    Math.round((num / total) * 100)
-
-  getStudentRowData: ->
-    # The period may not have been selected. If not, just use the 1st period
-    {sort, period_id, firstDataColumn, displayAs} = @state
-
-    data = ScoresStore.get(@props.courseId)
-    scores = if period_id
-      _.findWhere(data, {period_id})
+    asc = if @state.sort.key is key then (not @state.sort.asc) else false
+    @updateStudentData({sort: {key, asc, dataType}})
+  selectPeriod: (period) -> @updateStudentData({period_id: period.id})
+  setPeriodIndex:  (key) -> @updateStudentData({periodIndex: key + 1})
+  changeDisplayAs:(mode) -> @updateStudentData({displayAs: mode})
```

Comment:
yeah but it has to.  `displayAs` influences how values are sorted.  also, merge should be pretty cheap since it's not a complex deeply nested structure.